pkg/gofr/cmd/terminal/spinner.go

```diff
@@ -0,0 +1,80 @@
+package terminal
+
+import (
+	"time"
+)
+
+// Spinner is a TUI component that displays a loading spinner which can be used to
+// denote a running background process.
+type Spinner struct {
+	// frames denote the frames of the spinner that displays in continiuation
+	frames []string
+	// fps is the speed at which the spinner frames are displayed
+	fps time.Duration
+	// outStream is the Output stream to which the spinner frames are printed onto
+	outStream Output
+
+	// started denotes whether the spinner has started spinning and ticker
+	// is the time.Ticker for the continuous time update for the spinner.
+	started bool
+	ticker  *time.Ticker
+}
+
+func NewDotSpinner(o Output) *Spinner {
+	return &Spinner{
+		frames:    []string{"⣾ ", "⣽ ", "⣻ ", "⢿ ", "⡿ ", "⣟ ", "⣯ ", "⣷ "},
+		fps:       time.Second / 10,
+		outStream: o,
+	}
+}
+
+func NewPulseSpinner(o Output) *Spinner {
+	return &Spinner{
+		frames:    []string{"█", "▓", "▒", "░"},
+		fps:       time.Second / 4,
+		outStream: o,
+	}
+}
+
+func NewGlobeSpinner(o Output) *Spinner {
+	return &Spinner{
+		frames:    []string{"🌍", "🌎", "🌏"},
+		fps:       time.Second / 4,
+		outStream: o,
+	}
+}
+
+func (s *Spinner) Spin() *Spinner {
+	t := time.NewTicker(s.fps)
```

Comment:
I'm a bit worried a ticker could be created, but never cleaned,if no one call Stop

I'm unsure about the best way,but I would to raise a possible memory leak

Reply:
Could having a context in Spin() function and stopping the ticker on context.Done resolve this issue?

Reply:
And when someone does not call Stop() the spinner will always spin and will never stop as the control flow will not be passed back to the main go routine.

Reply:
Closing the ticker when the context is done.